proto/buf/alpha/registry/v1alpha1/reference.proto

```diff
@@ -86,11 +92,37 @@ message ListGitCommitMetadataForReferenceRequest {
   string owner = 2;
   // Name of the repository the reference belongs to.
   string repository_name = 3;
+  uint32 page_size = 4;
+  // The first page is returned if this is empty.
+  string page_token = 5;
 }
 
 message ListGitCommitMetadataForReferenceResponse {
   // ID of the BSR commit the reference resolved to.
   string commit_id = 1;
   // List of git commits and metadata associated with the resolved reference.
   repeated GitCommitMetadata git_commit_metadatas = 2;
+  // There are no more pages if this is empty.
+  string next_page_token = 3;
+}
+
+message ListGitCommitsNoMetadataForReferenceRequest {
+  // String that represents the name of the reference.
+  string reference = 1;
+  // Owner of the repository the reference belongs to.
+  string owner = 2;
+  // Name of the repository the reference belongs to.
+  string repository_name = 3;
+  uint32 page_size = 4;
+  // The first page is returned if this is empty.
+  string page_token = 5;
+}
+
+message ListGitCommitsNoMetadataForReferenceResponse {
+  // ID of the BSR commit the reference resolved to.
+  string commit_id = 1;
+  // List of git commits labels without metadata associated with the resolved reference.
+  repeated Label git_commit_labels_no_metadatas = 2;
```

Comment:
Nit: we already know from the parent message that these labels come from a commit that doesn't have git metadata on it.
```suggestion
  // List of git commits labels without metadata associated with the resolved reference.
  repeated Label git_commit_labels = 2;
```